app/code/Magento/Backup/Model/BackupFactory.php

```diff
@@ -41,20 +41,18 @@ public function create($timestamp, $type)
         $backupId = $timestamp . '_' . $type;
         $fsCollection = $this->_objectManager->get(\Magento\Backup\Model\Fs\Collection::class);
         $backupInstance = $this->_objectManager->get(\Magento\Backup\Model\Backup::class);
+
         foreach ($fsCollection as $backup) {
-            if ($backup->getId() == $backupId) {
-                $backupInstance->setType(
-                    $backup->getType()
-                )->setTime(
-                    $backup->getTime()
-                )->setName(
-                    $backup->getName()
-                )->setPath(
-                    $backup->getPath()
-                );
+            if (strpos($backup->getId(), $backupId) !== false) {
```

Comment:
Can we replace this condition with more accurate one? In which case previous `$backup->getId() == $backupId` condition didn't work?

For instance, if the string contained additional slash, we should search for `'/' . $backupId` and so on.